### ISSUE TITLE
Process freezes on mate

### ISSUE THREAD

**Issue by Heiaha** [Heiaha/asyncLio-bot#1]
I've noticed that the game seems to freeze [here](https://github.com/Heiaha/asyncLio-bot/blob/master/game.py#L129) every once in a while on a mate score returned, as in [this game](https://lichess.org/iLfk9l7u/black):

```
2022-07-29 15:00:21.481 | INFO     | game:make_move:145 - iLfk9l7u -- Searching for move from r5k1/1q3pp1/pB5p/P3p1N1/1b2r2P/1P6/5Pb1/1Q1R2K1 b - - 0 28.
2022-07-29 15:00:26.873 | INFO     | game:make_move:158 - iLfk9l7u -- Engine: 28...  hxg5      CP Score: +860      Time: 5.4       Depth: 18        PV: 28...hxg5 29. Kxg2
2022-07-29 15:00:34.685 | INFO     | game:make_move:145 - iLfk9l7u -- Searching for move from r5k1/1q3pp1/pB6/P3p1p1/1b2r2P/1P6/5PK1/1Q1R4 b - - 0 29.
2022-07-29 15:00:39.593 | INFO     | game:make_move:158 - iLfk9l7u -- Engine: 29...  Rf4+      CP Score: +970      Time: 4.9       Depth: 19        PV: 29...Rf4+ 30. Kf1 Qh1+ 31. Ke2 Qf3+ 32. Kf1 Bxa5 33. hxg5
2022-07-29 15:00:42.406 | INFO     | game:make_move:145 - iLfk9l7u -- Searching for move from r5k1/1q3pp1/pB6/P3p1p1/1b3r1P/1P6/5P1K/1Q1R4 b - - 2 30.
2022-07-29 15:06:42.066 | INFO     | game:_play:277 - BOT Boris-Trapsky won! Weiawaga ran out of time.
2022-07-29 15:06:42.083 | INFO     | game_manager:on_game_finish:93 - Current Processes: 0.
```

Sometimes it returns the mate score the instant the loop returns that it has run out of time, sometimes it doesn't (as seen here). It seems to be very difficult to reproduce. I'm not sure if this could potentially be the asyncio loop not returning the move every once in a while? Trying to reproduce the problem proves to be very difficult, as neither [starting from the position](https://lichess.org/hOkDgUArMfsg) with the bot bridge nor running the bot locally seem to reproduce the problem when it happens:

```
Weiawaga v5.0.0
https://github.com/Heiaha/Weiawaga
position fen r5k1/1q3pp1/pB6/P3p1p1/1b3r1P/1P6/5P1K/1Q1R4 b - - 2 30
go wtime 321000 btime 354000
info currmove g5h4 depth 1 seldepth 3 time 0 score cp 591 nodes 62 nps 62000 hashfull 0 pv g5h4 
info currmove f4h4 depth 2 seldepth 5 time 0 score cp 616 lowerbound nodes 79 nps 79000 hashfull 0 pv f4h4 
info currmove f4h4 depth 2 seldepth 5 time 0 score cp 616 nodes 375 nps 375000 hashfull 1 pv f4h4 h2g3 
info currmove f4h4 depth 3 seldepth 8 time 0 score cp 613 nodes 1191 nps 1191000 hashfull 1 pv f4h4 h2g3 b7e4 
info currmove f4h4 depth 4 seldepth 8 time 0 score cp 638 lowerbound nodes 145 nps 145000 hashfull 1 pv f4h4 
info currmove b4a5 depth 4 seldepth 8 time 1 score cp 665 nodes 1705 nps 852500 hashfull 1 pv b4a5 h4g5 
info currmove b4a5 depth 5 seldepth 8 time 2 score cp 663 nodes 2975 nps 991666 hashfull 1 pv b4a5 b6e3 f4h4 h2g3 a5c7 
info currmove f4g4 depth 6 seldepth 11 time 3 score cp 688 lowerbound nodes 6098 nps 1524500 hashfull 1 pv f4g4 
info currmove f4g4 depth 6 seldepth 11 time 4 score cp 688 nodes 7526 nps 1505200 hashfull 1 pv f4g4 b1h7 
info currmove b4a5 depth 7 seldepth 13 time 9 score cp 712 nodes 17033 nps 1703300 hashfull 2 pv b4a5 b6e3 f4g4 
info currmove b4a5 depth 8 seldepth 11 time 10 score cp 737 lowerbound nodes 4721 nps 429181 hashfull 2 pv b4a5 
info currmove b7f3 depth 8 seldepth 15 time 16 score cp 738 nodes 28859 nps 1697588 hashfull 4 pv b7f3 h2g1 
info currmove b7f3 depth 9 seldepth 13 time 17 score cp 763 lowerbound nodes 2269 nps 126055 hashfull 4 pv b7f3 
info currmove b7f3 depth 9 seldepth 15 time 24 score cp 763 nodes 26735 nps 1069400 hashfull 4 pv b7f3 h2g1 
info currmove b7f3 depth 10 seldepth 13 time 24 score cp 788 lowerbound nodes 2755 nps 110200 hashfull 4 pv b7f3 
info currmove b7f3 depth 10 seldepth 16 time 41 score cp 788 nodes 65872 nps 1568380 hashfull 5 pv b7f3 h2g1 
info currmove b7f3 depth 11 seldepth 14 time 42 score cp 813 lowerbound nodes 4564 nps 106139 hashfull 6 pv b7f3 
info currmove b7f3 depth 11 seldepth 18 time 57 score cp 813 nodes 59065 nps 1018362 hashfull 9 pv b7f3 h2g1 
info currmove b7f3 depth 12 seldepth 21 time 59 score cp 838 lowerbound nodes 8949 nps 149150 hashfull 10 pv b7f3 
info currmove b7f3 depth 12 seldepth 22 time 86 score cp 838 nodes 116614 nps 1340390 hashfull 20 pv b7f3 h2g1 
info currmove b7f3 depth 13 seldepth 19 time 90 score cp 863 lowerbound nodes 13982 nps 153648 hashfull 23 pv b7f3 
info currmove f7f5 depth 13 seldepth 21 time 205 score cp 864 nodes 453400 nps 2200970 hashfull 56 pv f7f5 h2g1 
info currmove f7f5 depth 14 seldepth 21 time 288 score cp 889 lowerbound nodes 272672 nps 943501 hashfull 78 pv f7f5 
info currmove b4d2 depth 14 seldepth 24 time 757 score cp 893 nodes 1979419 nps 2611370 hashfull 224 pv b4d2 d1d2 
info currmove f4h4 depth 15 seldepth 21 time 783 score cp 918 lowerbound nodes 91845 nps 117149 hashfull 232 pv f4h4 
info currmove f4h4 depth 15 seldepth 21 time 979 score mate 7 nodes 869667 nps 887415 hashfull 304 pv f4h4 h2g3 h4g4 g3g4 b7g2 g4f5 g2f3 f5e5 a8e8 
bestmove f4h4
```

**Comment by Heiaha** [comment]
The bug may have been fixed by more robust retrying on http connection errors in d40bb5c7196d27418495538a33cb86dce2cb81b5. I will do further testing since this error was so difficult to reproduce.

**Comment by Heiaha** [comment]
Problem still seems to occur, as can be seen in [this game](https://lichess.org/0ZgEgBb0/white#128). Logs seem to indicate that the problem happens on 

```
result = await self.engine.play(self.board, limit, info=chess.engine.INFO_ALL)
```
As seen here: 
```
2022-08-02 16:24:56.898 | DEBUG    | lichess:watch_game_stream:104 - Game event: {'type': 'gameState', 'moves': 'd2d4 d7d5 g1f3 g8f6 c2c4 d5c4 e2e3 e7e6 f1c4 a7a6 a2a4 c7c5 e1g1 b8c6 d1e2 d8c7 f1d1 f8e7 d4c5 e6e5 b1c3 e8g8 c3d5 f6d5 c4d5 c6b4 e3e4 b4d5 e4d5 e7c5 e2e5 c5d6 e5c3 c7c3 b2c3 c8g4 c1a3 g4f3 g2f3 f8d8 a3d6 d8d6 d1d4 a8c8 c3c4 a6a5 a1b1 b7b6 g1f1 c8e8 b1b5 e8c8 f1e2 g7g6 e2d3 g8g7 d4e4 d6f6 f3f4 c8d8 d3d4 d8d6 c4c5 b6c5 d4c5 d6d7 b5a5 d7c7 c5b5 c7b7 b5c4 b7b2 e4d4 f6d6 c4c5 d6d8 d5d6 d8c8 c5d5 c8d8 d5c6 d8c8 c6d7 b2b8 a5a7 c8c2 a4a5 b8b1 d7e7 c2e2 e7d8 b1b8 d8c7 b8h8 a7a6 e2e6 a6b6 e6e8 d6d7 e8a8 d4b4 a8a5 b6b8 h8b8 b4b8 a5d5 b8b6 d5d7 c7d7 g7h6 b6b3 g6g5 f4f5 g5g4 d7e7 h6g5 f5f6 g5g6 b3g3 h7h6 h2h4 h6h5 g3g1 g6f5 e7f7 f5f4 g1g3 f4e4', 'wtime': 76110, 'btime': 60430, 'winc': 0, 'binc': 0, 'status': 'started'}
2022-08-02 16:24:56.915 | INFO     | game:make_move:145 - 0ZgEgBb0 -- Searching for move from 8/5K2/5P2/7p/4k1pP/6R1/5P2/8 w - - 3 65.
2022-08-02 16:26:15.645 | INFO     | game:make_move:158 - 0ZgEgBb0 -- Engine: 65.  Ke6       Mate: +6        Time: 0.2       Depth: 13        PV: 65. Ke6 Kd4 66. f7 Kc4 67. f8=Q
2022-08-02 16:26:15.646 | DEBUG    | lichess:post:71 - /api/bot/game/0ZgEgBb0/move/f7e6: {'params': {'offeringDraw': 'false'}}.
2022-08-02 16:26:15.855 | DEBUG    | lichess:watch_event_stream:86 - Event: {'type': 'gameFinish', 'game': {'fullId': '0ZgEgBb0Ac2U', 'gameId': '0ZgEgBb0', 'fen': '8/5K2/5P2/7p/4k1pP/6R1/5P2/8 w - - 3 65', 'color': 'white', 'lastMove': 'f4e4', 'source': 'friend', 'variant': {'key': 'standard', 'name': 'Standard'}, 'speed': 'rapid', 'perf': 'rapid', 'rated': True, 'hasMoved': True, 'opponent': {'id': 'inanisbot', 'username': 'BOT InanisBot', 'rating': 2364}, 'isMyTurn': False, 'secondsLeft': 0, 'compat': {'bot': True, 'board': True}, 'id': '0ZgEgBb0'}}
2022-08-02 16:26:15.859 | DEBUG    | lichess:watch_game_stream:104 - Game event: {'type': 'gameState', 'moves': 'd2d4 d7d5 g1f3 g8f6 c2c4 d5c4 e2e3 e7e6 f1c4 a7a6 a2a4 c7c5 e1g1 b8c6 d1e2 d8c7 f1d1 f8e7 d4c5 e6e5 b1c3 e8g8 c3d5 f6d5 c4d5 c6b4 e3e4 b4d5 e4d5 e7c5 e2e5 c5d6 e5c3 c7c3 b2c3 c8g4 c1a3 g4f3 g2f3 f8d8 a3d6 d8d6 d1d4 a8c8 c3c4 a6a5 a1b1 b7b6 g1f1 c8e8 b1b5 e8c8 f1e2 g7g6 e2d3 g8g7 d4e4 d6f6 f3f4 c8d8 d3d4 d8d6 c4c5 b6c5 d4c5 d6d7 b5a5 d7c7 c5b5 c7b7 b5c4 b7b2 e4d4 f6d6 c4c5 d6d8 d5d6 d8c8 c5d5 c8d8 d5c6 d8c8 c6d7 b2b8 a5a7 c8c2 a4a5 b8b1 d7e7 c2e2 e7d8 b1b8 d8c7 b8h8 a7a6 e2e6 a6b6 e6e8 d6d7 e8a8 d4b4 a8a5 b6b8 h8b8 b4b8 a5d5 b8b6 d5d7 c7d7 g7h6 b6b3 g6g5 f4f5 g5g4 d7e7 h6g5 f5f6 g5g6 b3g3 h7h6 h2h4 h6h5 g3g1 g6f5 e7f7 f5f4 g1g3 f4e4', 'wtime': 0, 'btime': 60430, 'winc': 0, 'binc': 0, 'status': 'outoftime', 'winner': 'black'}
2022-08-02 16:26:15.861 | INFO     | game:_play:277 - BOT InanisBot won! Weiawaga ran out of time.
```

Note the time stamps between lines 2 and 3.

**Comment by Heiaha** [comment]
Another instance, this time with better logging by using the default logging in python-chess:

```
2022-08-04 00:23:20 INFO     TK9VRWav -- Searching for move from 7Q/1b2bk2/2q1p3/p7/Ppp2R2/4PP2/1P4PP/6K1 b - - 5 37.
2022-08-04 00:23:20 DEBUG    Starting move search...
2022-08-04 00:23:20 DEBUG    <UciProtocol (pid=56111)>: << position startpos moves d2d4 g8f6 c2c4 e7e6 g1f3 d7d5 b1c3 c7c6 c1g5 f8e7 e2e3 e8g8 a1c1 b8d7 f1d3 h7h6 g5h4 d5c4 d3c4 b7b5 c4d3 a7a6 a2a4 c8b7 e1g1 b5b4 h4f6 e7f6 c3e4 f6e7 d3b1 a6a5 f3e5 d7e5 d4e5 c6c5 d1g4 d8c7 e4f6 g8h8 g4h5 c5c4 f6g4 c7c6 f2f3 h8g8 f1d1 a8d8 g4h6 g7h6 h5h6 f7f5 e5f6 d8d1 c1d1 f8f6 b1h7 g8f7 h6h5 f7g7 d1d4 g7f8 h5g5 f8e8 h7g6 f6f7 g6f7 e8f7 g5h5 f7f8 h5h8 f8f7 d4f4
2022-08-04 00:23:20 DEBUG    <UciProtocol (pid=56111)>: << go wtime 296820 btime 19150 winc 1000 binc 1000
2022-08-04 00:23:20 DEBUG    <UciProtocol (pid=56111)>: >> info currmove f7g6 depth 1 seldepth 4 time 0 score cp -64 nodes 65 nps 65000 hashfull 0 pv f7g6 
2022-08-04 00:23:20 DEBUG    <UciProtocol (pid=56111)>: >> info currmove f7g6 depth 2 seldepth 6 time 0 score cp -89 upperbound nodes 88 nps 88000 hashfull 0 pv f7g6 h8g8 
2022-08-04 00:23:20 DEBUG    <UciProtocol (pid=56111)>: >> info currmove f7g6 depth 2 seldepth 7 time 0 score cp -119 nodes 180 nps 180000 hashfull 0 pv f7g6 h8g8 
2022-08-04 00:23:20 DEBUG    <UciProtocol (pid=56111)>: >> info currmove f7g6 depth 3 seldepth 10 time 1 score cp -111 nodes 403 nps 201500 hashfull 0 pv f7g6 h8g8 g6h6 
2022-08-04 00:23:20 DEBUG    <UciProtocol (pid=56111)>: >> info currmove f7g6 depth 4 seldepth 10 time 1 score cp -136 upperbound nodes 116 nps 58000 hashfull 0 pv f7g6 h8g8 
2022-08-04 00:23:20 DEBUG    <UciProtocol (pid=56111)>: >> info currmove f7g6 depth 4 seldepth 10 time 1 score cp -136 nodes 792 nps 396000 hashfull 0 pv f7g6 
2022-08-04 00:23:20 DEBUG    <UciProtocol (pid=56111)>: >> info currmove f7g6 depth 5 seldepth 12 time 2 score cp -161 upperbound nodes 781 nps 260333 hashfull 0 pv f7g6 h8g8 
2022-08-04 00:23:20 DEBUG    <UciProtocol (pid=56111)>: >> info currmove f7g6 depth 5 seldepth 12 time 5 score cp -394 nodes 3812 nps 635333 hashfull 0 pv f7g6 f4g4 e7g5 h8e5 g6h6 
2022-08-04 00:23:20 DEBUG    <UciProtocol (pid=56111)>: >> info currmove f7g6 depth 6 seldepth 15 time 8 score cp -419 upperbound nodes 2761 nps 306777 hashfull 0 pv f7g6 f4g4 
2022-08-04 00:23:20 DEBUG    <UciProtocol (pid=56111)>: >> info currmove f7g6 depth 6 seldepth 17 time 15 score cp -479 nodes 9517 nps 594812 hashfull 0 pv f7g6 f4g4 e7g5 h8g8 g6f6 g8g5 
2022-08-04 00:23:20 DEBUG    <UciProtocol (pid=56111)>: >> info currmove f7g6 depth 7 seldepth 16 time 16 score cp -504 upperbound nodes 1135 nps 66764 hashfull 0 pv f7g6 f4g4 
2022-08-04 00:23:20 DEBUG    <UciProtocol (pid=56111)>: >> info currmove f7g6 depth 7 seldepth 18 time 22 score cp -925 nodes 7857 nps 341608 hashfull 2 pv f7g6 f4g4 g6f5 e3e4 c6e4 h8h7 f5e5 
2022-08-04 00:23:20 DEBUG    <UciProtocol (pid=56111)>: >> info currmove f7g6 depth 8 seldepth 16 time 23 score cp -950 upperbound nodes 714 nps 29750 hashfull 2 pv f7g6 f4g4 
2022-08-04 00:23:20 DEBUG    <UciProtocol (pid=56111)>: >> info currmove f7g6 depth 8 seldepth 18 time 24 score cp -950 nodes 2879 nps 115160 hashfull 2 pv f7g6 
2022-08-04 00:23:20 DEBUG    <UciProtocol (pid=56111)>: >> info currmove f7g6 depth 9 seldepth 16 time 29 score cp -925 lowerbound nodes 4946 nps 164866 hashfull 2 pv f7g6 
2022-08-04 00:23:20 DEBUG    <UciProtocol (pid=56111)>: >> info currmove f7g6 depth 9 seldepth 21 time 49 score cp -925 nodes 25726 nps 514520 hashfull 3 pv f7g6 f4g4 
2022-08-04 00:23:20 DEBUG    <UciProtocol (pid=56111)>: >> info currmove f7g6 depth 10 seldepth 21 time 58 score cp -950 upperbound nodes 10026 nps 169932 hashfull 3 pv f7g6 f4g4 e7g5 
2022-08-04 00:23:20 DEBUG    <UciProtocol (pid=56111)>: >> info currmove f7g6 depth 10 seldepth 21 time 59 score cp -950 nodes 11062 nps 184366 hashfull 3 pv f7g6 
2022-08-04 00:23:20 DEBUG    <UciProtocol (pid=56111)>: >> info currmove f7g6 depth 11 seldepth 19 time 61 score cp -975 upperbound nodes 1337 nps 21564 hashfull 3 pv f7g6 f4g4 
2022-08-04 00:23:20 DEBUG    <UciProtocol (pid=56111)>: >> info currmove f7g6 depth 11 seldepth 21 time 65 score cp -976 nodes 6332 nps 95939 hashfull 3 pv f7g6 
2022-08-04 00:23:20 DEBUG    <UciProtocol (pid=56111)>: >> info currmove f7g6 depth 12 seldepth 21 time 72 score cp -1001 upperbound nodes 6636 nps 90904 hashfull 3 pv f7g6 f4g4 
2022-08-04 00:23:20 DEBUG    <UciProtocol (pid=56111)>: >> info currmove f7g6 depth 12 seldepth 21 time 79 score cp -1001 nodes 14455 nps 180687 hashfull 4 pv f7g6 
2022-08-04 00:23:20 DEBUG    <UciProtocol (pid=56111)>: >> info currmove f7g6 depth 13 seldepth 22 time 84 score cp -1026 upperbound nodes 4577 nps 53847 hashfull 4 pv f7g6 f4g4 
2022-08-04 00:23:20 DEBUG    <UciProtocol (pid=56111)>: >> info currmove f7g6 depth 13 seldepth 22 time 103 score cp -1027 nodes 26431 nps 254144 hashfull 5 pv f7g6 h8g8 g6h6 
2022-08-04 00:23:20 DEBUG    <UciProtocol (pid=56111)>: >> info currmove f7g6 depth 14 seldepth 22 time 113 score cp -1052 upperbound nodes 9107 nps 79885 hashfull 5 pv f7g6 h8g8 
2022-08-04 00:23:20 DEBUG    <UciProtocol (pid=56111)>: >> info currmove f7g6 depth 14 seldepth 26 time 184 score cp -1054 nodes 87127 nps 470956 hashfull 7 pv f7g6 f4g4 e7g5 g4g5 g6g5 
2022-08-04 00:23:21 DEBUG    <UciProtocol (pid=56111)>: >> info currmove f7g6 depth 15 seldepth 28 time 238 score cp -1079 upperbound nodes 59530 nps 249079 hashfull 9 pv f7g6 f4g4 
2022-08-04 00:23:21 DEBUG    <UciProtocol (pid=56111)>: >> info currmove f7g6 depth 15 seldepth 28 time 283 score cp -1079 nodes 112252 nps 395253 hashfull 11 pv f7g6 
2022-08-04 00:23:21 DEBUG    <UciProtocol (pid=56111)>: >> info currmove f7g6 depth 16 seldepth 29 time 384 score cp -1104 upperbound nodes 106670 nps 277064 hashfull 16 pv f7g6 f4g4 
2022-08-04 00:23:21 DEBUG    <UciProtocol (pid=56111)>: >> info currmove f7g6 depth 16 seldepth 29 time 541 score cp -1104 nodes 288675 nps 532610 hashfull 24 pv f7g6 
2022-08-04 00:23:21 DEBUG    <UciProtocol (pid=56111)>: >> info currmove f7g6 depth 17 seldepth 32 time 646 score cp -1129 upperbound nodes 107903 nps 166774 hashfull 26 pv f7g6 f4g4 
2022-08-04 00:23:21 DEBUG    <UciProtocol (pid=56111)>: >> info currmove f7g6 depth 17 seldepth 32 time 849 score cp -1129 nodes 338992 nps 398814 hashfull 30 pv f7g6 
2022-08-04 00:23:21 DEBUG    <UciProtocol (pid=56111)>: >> info currmove f7g6 depth 18 seldepth 32 time 1010 score cp -1154 upperbound nodes 176050 nps 174134 hashfull 38 pv f7g6 f4g4 
2022-08-04 00:23:22 DEBUG    <UciProtocol (pid=56111)>: >> info currmove f7g6 depth 18 seldepth 32 time 1240 score mate -5 nodes 461088 nps 371545 hashfull 50 pv f7g6 f4g4 e7g5 g1h1 c6c5 h2h4 g6f7 g4g5 c5f8 h8h7 f7f6 h7b7 f8h8 g5a5 h8h4 h1g1 
2022-08-04 00:23:49 DEBUG    Event: {'type': 'gameFinish', 'game': {'fullId': 'TK9VRWavmEVe', 'gameId': 'TK9VRWav', 'fen': '7Q/1b2bk2/2q1p3/p7/Ppp2R2/4PP2/1P4PP/6K1 b - - 5 37', 'color': 'black', 'lastMove': 'd4f4', 'source': 'friend', 'variant': {'key': 'standard', 'name': 'Standard'}, 'speed': 'rapid', 'perf': 'rapid', 'rated': False, 'hasMoved': True, 'opponent': {'id': 'opulent_ophelia', 'username': 'BOT Opulent_Ophelia', 'rating': 2564}, 'isMyTurn': False, 'secondsLeft': 0, 'compat': {'bot': True, 'board': True}, 'id': 'TK9VRWav'}}
```
So, from the logs it looks like the engine never returns a "bestmove", but when I run locally from the same position by hand on the same machine it seems to return bestmove with no issue:

```
Weiawaga v5.0.0
https://github.com/Heiaha/Weiawaga
setoption name Threads value 3
info string set Threads to 3
setoption name Hash value 128
info string set Hash to 128MB
position startpos moves d2d4 g8f6 c2c4 e7e6 g1f3 d7d5 b1c3 c7c6 c1g5 f8e7 e2e3 e8g8 a1c1 b8d7 f1d3 h7h6 g5h4 d5c4 d3c4 b7b5 c4d3 a7a6 a2a4 c8b7 e1g1 b5b4 h4f6 e7f6 c3e4 f6e7 d3b1 a6a5 f3e5 d7e5 d4e5 c6c5 d1g4 d8c7 e4f6 g8h8 g4h5 c5c4 f6g4 c7c6 f2f3 h8g8 f1d1 a8d8 g4h6 g7h6 h5h6 f7f5 e5f6 d8d1 c1d1 f8f6 b1h7 g8f7 h6h5 f7g7 d1d4 g7f8 h5g5 f8e8 h7g6 f6f7 g6f7 e8f7 g5h5 f7f8 h5h8 f8f7 d4f4
go wtime 296820 btime 19150 winc 1000 binc 1000
info currmove f7g6 depth 1 seldepth 4 time 0 score cp -64 nodes 65 nps 65000 hashfull 0 pv f7g6
info currmove f7g6 depth 2 seldepth 6 time 1 score cp -89 upperbound nodes 107 nps 53500 hashfull 0 pv f7g6 h8g8
info currmove f7g6 depth 2 seldepth 6 time 1 score cp -119 nodes 196 nps 98000 hashfull 0 pv f7g6 h8g8
info currmove f7g6 depth 3 seldepth 10 time 3 score cp -111 nodes 357 nps 89250 hashfull 0 pv f7g6 h8g8 g6h6
info currmove f7g6 depth 4 seldepth 10 time 3 score cp -136 upperbound nodes 121 nps 30250 hashfull 0 pv f7g6 h8g8
info currmove f7g6 depth 4 seldepth 10 time 6 score cp -136 nodes 806 nps 115142 hashfull 0 pv f7g6
info currmove f7g6 depth 5 seldepth 12 time 8 score cp -161 upperbound nodes 712 nps 79111 hashfull 0 pv f7g6 h8g8
info currmove f7g6 depth 5 seldepth 15 time 14 score cp -468 nodes 4999 nps 333266 hashfull 2 pv f7g6 f4g4 e7g5 h2h4 c4c3
info currmove f7g6 depth 6 seldepth 16 time 19 score cp -470 nodes 4469 nps 223450 hashfull 2 pv f7g6 f4g4 e7g5 h8g8 g6f6 g8g5
info currmove f7g6 depth 7 seldepth 17 time 21 score cp -495 upperbound nodes 2211 nps 100500 hashfull 2 pv f7g6 f4g4
info currmove f7g6 depth 7 seldepth 18 time 27 score cp -925 nodes 9514 nps 339785 hashfull 3 pv f7g6 f4g4 g6f5 e3e4 c6e4 h8h7 f5e5
info currmove f7g6 depth 8 seldepth 16 time 28 score cp -950 upperbound nodes 310 nps 10689 hashfull 3 pv f7g6 f4g4
info currmove f7g6 depth 8 seldepth 16 time 29 score cp -950 nodes 2043 nps 68100 hashfull 3 pv f7g6
info currmove f7g6 depth 9 seldepth 16 time 34 score cp -925 lowerbound nodes 5858 nps 167371 hashfull 3 pv f7g6
info currmove f7g6 depth 9 seldepth 24 time 85 score cp -925 nodes 57041 nps 663267 hashfull 5 pv f7g6 f4g4
info currmove f7g6 depth 10 seldepth 17 time 86 score cp -950 upperbound nodes 65 nps 747 hashfull 5 pv f7g6 f4g4
info currmove f7g6 depth 10 seldepth 17 time 92 score cp -951 nodes 6934 nps 74559 hashfull 5 pv f7g6 h8g8 g6h6
info currmove f7g6 depth 11 seldepth 21 time 97 score cp -976 upperbound nodes 5713 nps 58295 hashfull 5 pv f7g6 f4g4
info currmove f7g6 depth 11 seldepth 21 time 102 score cp -976 nodes 11715 nps 113737 hashfull 5 pv f7g6
info currmove f7g6 depth 12 seldepth 22 time 113 score cp -1001 upperbound nodes 11508 nps 100947 hashfull 5 pv f7g6 f4g4
info currmove f7g6 depth 12 seldepth 22 time 140 score mate -5 nodes 42450 nps 301063 hashfull 5 pv f7g6 f4g4 e7g5 g4g5 g6g5 h8g7 g5h5 g2g4 h5h4 g7h6
bestmove f7g6
```

Given Weiawaga's code, it's not clear to me how the search could go so long without printing anything at all. I will test by running some games with Stockfish to see if I can replicate this and eliminate the possibility that it may be specific to Weiawaga.

**Comment by Heiaha** [comment]
This seems to be a Weiawaga specific bug not associated with asyncLio-bot. It is likely fixed by 49712ccb9ce9b7086d6ad8d664e6d8c8ad9e8d46. Will do some more testing and then close this issue if it appears to be fixed.
